src/mpi/group/group_impl.c

```diff
@@ -674,8 +677,46 @@ int MPIR_Group_from_session_pset_impl(MPIR_Session * session_ptr, const char *ps
         group_ptr->lrank_to_lpid[0].next_lpid = -1;
         group_ptr->idx_of_first_lpid = 0;
     } else {
-        /* TODO: Implement pset struct, locate pset struct ptr */
-        MPIR_ERR_SETANDSTMT(mpi_errno, MPI_ERR_ARG, goto fn_fail, "**psetinvalidname");
+        MPIR_Pset *pset = NULL;
+        mpi_errno = MPIR_Pset_by_name(pset_name, &pset);
+
+        if (mpi_errno == MPI_ERR_OTHER || pset == NULL) {
+            *new_group_ptr = NULL;
+            MPIR_ERR_SETANDSTMT(mpi_errno, MPI_ERR_ARG, goto fn_fail, "**psetinvalidname");
+        }
+
+        /* Remark: We do not check the pset for validity here. Creating a Process Group
+         * from an invalid pset should be possible as a local operation. However, creating
+         * an MPI communicator from a Process Group that is based on an invalid pset
+         * should raise an error. */
+
+        mpi_errno = MPIR_Group_create(pset->size, &group_ptr);
+        MPIR_ERR_CHECK(mpi_errno);
+
```

Comment:
Here is the 2nd point I was hanging at. The current Group facility is based on `lpid`, which is kinda opaque and controlled by the device layer. So what we needed here is an MPID API that translates from `struct pset_member` to `lpid`. With that, we can construct a `MPIR_Group` from a `MPIR_Pset`, then continue to create communicator using the existing Group facilities.

Reply:
Mhh.. I need to think about this a bit more.

But shouldn't this issue be resolved if the mapping of PMIx proc ID (aka `pset_member` in your comment) to MPI rank/ `lpid` happens in `mpir_pmix`?